source/checksums.c

```diff
@@ -55,9 +53,69 @@ napi_value crc_common(napi_env env, napi_callback_info info, uint32_t (*checksum
 }
 
 napi_value aws_napi_checksums_crc32(napi_env env, napi_callback_info info) {
-    return crc_common(env, info, aws_checksums_crc32);
+    return crc32_common(env, info, aws_checksums_crc32_ex);
 }
 
 napi_value aws_napi_checksums_crc32c(napi_env env, napi_callback_info info) {
-    return crc_common(env, info, aws_checksums_crc32c);
+    return crc32_common(env, info, aws_checksums_crc32c_ex);
+}
+
+napi_value aws_napi_checksums_crc64nvme(napi_env env, napi_callback_info info) {
+    napi_value node_args[2];
+    size_t num_args = AWS_ARRAY_SIZE(node_args);
+    struct aws_byte_buf to_hash;
+    AWS_ZERO_STRUCT(to_hash);
+    struct aws_byte_buf previous_buf;
+    AWS_ZERO_STRUCT(previous_buf);
+
+    napi_value node_val = NULL;
+
+    if (napi_get_cb_info(env, info, &num_args, node_args, NULL, NULL)) {
+        napi_throw_error(env, NULL, "Failed to retrieve callback information");
+        goto done;
+    }
+    if (num_args != AWS_ARRAY_SIZE(node_args)) {
+        napi_throw_error(env, NULL, "aws_napi_checksums_crc64 needs exactly 2 arguments");
+        goto done;
+    }
+
+    if (aws_byte_buf_init_from_napi(&to_hash, env, node_args[0])) {
+        napi_throw_type_error(env, NULL, "to_hash argument must be a string or array");
+        goto done;
+    }
+    uint8_t *buffer = to_hash.buffer;
+    size_t length = to_hash.len;
+
+    uint64_t previous = 0;
+    if (!aws_napi_is_null_or_undefined(env, node_args[1])) {
+        if (aws_byte_buf_init_from_napi(&previous_buf, env, node_args[1])) {
+            napi_throw_type_error(env, NULL, "previous argument must be undefined or a positive number");
```

Comment:
The `previous` is confusing to be a name of the variable or a real word. 

```suggestion
            napi_throw_type_error(env, NULL, "`previous` argument must be undefined or a positive number");
```

Also, a positive number doesn't seem to descriptive enough as a 64bit data? Oh, I guess you were copying from the crc32 case. So, we should update this.

Reply:
yeah, previous naming was copied from crc32 case. let me update the naming to indicate its previous crc